server/models/user.js

```diff
@@ -187,15 +193,23 @@ userSchema.statics.findByUsername = function findByUsername(username, cb) {
  * a username or email.
  *
  * @param {string} value - Email or username
+ * @param {boolean} caseInsensitive - Does a caseInsensitive query rather than
+ *                                    default query for username or email, defaults
+ *                                    to false
  * @callback [cb] - Optional error-first callback that passes User document
  * @return {Promise<Object>} - Returns Promise fulfilled by User document
  */
-userSchema.statics.findByEmailOrUsername = function findByEmailOrUsername(value, cb) {
+userSchema.statics.findByEmailOrUsername = function findByEmailOrUsername(value, caseInsensitive, cb) {
   const isEmail = value.indexOf('@') > -1;
```

Comment:
We [validate usernames not containing "@" chars in the client](https://github.com/processing/p5.js-web-editor/blob/7dbcde452b16df1d7d49c7bb83a9b630902e5846/client/utils/reduxFormUtils.js#L26) but I wonder whether this should also happen in the mongoose User model?

Reply:
Do you mean validating a username when saving a (new or existing) `User` document?

Reply:
Yes! Just to ensure these username rules are respected at the core of the app.